packaging/language/pip.py

```diff
@@ -147,13 +147,21 @@ def _get_cmd_options(module, cmd):
     words = stdout.strip().split()
     cmd_options = [ x for x in words if x.startswith('--') ]
     return cmd_options
-    
+
 
 def _get_full_name(name, version=None):
-    if version is None:
-        resp = name
-    else:
+    resp = name
+
+    if version is not None:
+        # this will support version='>1.0.1'
+        for op in ('>', '>=', '<', '<='):
+            # remove all white space so it can support version=' >1.0.1'
+            version = str(version).strip()
+            if version.startswith(op):
```

Comment:
Seems that this `startswith` in combination with the for loop is really only checking whether it starts with `>` or `<` due to the order of operations.  Perhaps we shouldn't even perform this check.  The version specs can be rather complex, and validating them won't be easy without using something like `pkg_resources` for doing so.

I'd vote for refactoring this to be more flexible and smarter.